plugins/languages/treesitter/treesitter.nix

```
@@ -6,210 +6,317 @@
+      "keymaps"
+      "initSelection"
+      "scopeIncremental"
+    ]
+  ];
 
-      gccPackage = helpers.mkPackageOption {
-        default = if cfg.nixGrammars then null else pkgs.gcc;
-        description = ''
-          Which package (if any) to be added as the GCC compiler.
-          This is required to build grammars if you are not using `nixGrammars`.
-          To disable the installation of GCC, set this option to `null`.
-        '';
-      };
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "treesitter"
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+    in
+    [
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "customCaptures" ]) (
+        settingsPath
+        ++ [
+          "highlight"
+          "custom_captures"
+        ]
+      ))
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "disabledLanguages" ]) (
+        settingsPath
+        ++ [
+          "highlight"
+          "disable"
+        ]
+      ))
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "indent" ]) (
+        settingsPath
+        ++ [
+          "indent"
+          "enable"
+        ]
+      ))
+    ];
+
+  settingsOptions = {
+    auto_install = helpers.defaultNullOpts.mkBool false ''
+      Whether to automatically install missing parsers when entering a buffer.
+    '';
 
-      parserInstallDir = mkOption {
-        type = types.nullOr types.str;
-        default = if cfg.nixGrammars then null else "$XDG_DATA_HOME/nvim/treesitter";
-        description = ''
-          Location of the parsers to be installed by the plugin (only needed when nixGrammars is disabled).
-          This default might not work on your own install, please make sure that $XDG_DATA_HOME is set if you want to use the default. Otherwise, change it to something that will work for you!
+    highlight = {
+      additional_vim_regex_highlighting =
+        with helpers.nixvimTypes;
+        helpers.defaultNullOpts.mkNullableWithRaw (either bool (listOf (maybeRaw str))) false ''
+          Setting this to true will run `:h syntax` and tree-sitter at the same time. \
+          Set this to `true` if you depend on 'syntax' being enabled (like for indentation). \
+          Using this option may slow down your editor, and you may see some duplicate highlights. \
+          Instead of true it can also be a list of languages
         '';
-      };
 
-      ignoreInstall = mkOption {
-        type = types.listOf types.str;
-        default = [ ];
-        description = "List of parsers to ignore installing (for \"all\")";
-      };
+      enable = helpers.defaultNullOpts.mkBool false ''
+        Whether to enable treesitter highlighting.
+      '';
 
-      disabledLanguages = mkOption {
-        type = types.listOf types.str;
-        default = [ ];
-        description = "A list of languages to disable";
-      };
+      disable =
+        with types;
+        helpers.defaultNullOpts.mkStrLuaFnOr (listOf str) "null" ''
```

```suggestion
        helpers.defaultNullOpts.mkStrLuaFnOr (listOf str) null ''
```

Unless you mean the _string_ `"null"`

---

No, sorry... I just thought I remembered a PR where we had to do a null string

---

Yes this changed only a few weeks ago